Lib/tarfile.py

```diff
@@ -883,6 +906,9 @@ class TarInfo(object):
         _link_target = None,
         )
 
+    _unames = {}    # Cached mappings of uid=0 -> uname
+    _gnames = {}    # Cached mappings of gid=0 -> gname
```

Comment:
I prefer that we keep per-instance caches instead of per-class caches, even for 0.

Reply:
Hello, sorry but there seems no available `TarFile` object for `_create_gnu_long_header` to cache the querying result:
- a `TarFile` instance does have cache members of `self._unames: Dict[uid, uname]`
- however, across the calling stack of `TarInfo.tobuf() -> TarInfo.create_gnu_header() -> TarInfo._create_gnu_long_header()`, there's no `TarFile` argument.

If we add a `TarFile` into `TarInfo.tobuf(...)`, then this PR may break existing subclasses of `TarInfo`. Is it indeed necessary?

Reply:
@picnixz Any idea about where to put the cache object?